tests/phpunit/elementor/includes/test-user.php

```diff
@@ -31,4 +32,206 @@ public function test_get_user_option_with_default__returns_value() {
+		$user = $this->act_as_admin();
+		$notices = [ 'test_notice' => [ 'is_viewed' => true ] ];
+
+		update_user_meta( $user->ID, self::ADMIN_NOTICES_KEY, $notices );
+
+		// Act.
+		$notices = User::get_user_notices();
+
+		// Assert.
+		$this->assertEquals( $notices, $notices );
+	}
+
+	public function test_set_user_notice__sets_new_notice() {
+		// Arrange.
+		$user = $this->act_as_admin();
+
+		// Act.
+		User::set_user_notice( 'test_notice' );
+
+		// Assert.
+		$notices = get_user_meta( $user->ID, self::ADMIN_NOTICES_KEY, true );
+		$this->assertEquals( [ 'test_notice' => [ 'is_viewed' => true, 'meta' => [] ] ], $notices );
+	}
+
+	public function test_set_user_notice__sets_existing_notice() {
+		// Arrange.
+		$user = $this->act_as_admin();
+		$notices = [ 'test_notice' => [ 'is_viewed' => true ] ];
+
+		update_user_meta( $user->ID, self::ADMIN_NOTICES_KEY, $notices );
+
+		// Act.
+		User::set_user_notice( 'test_notice' );
+
+		// Assert.
+		$notices = get_user_meta( $user->ID, self::ADMIN_NOTICES_KEY, true );
+		$this->assertEquals( [ 'test_notice' => [ 'is_viewed' => true, 'meta' => [] ] ], $notices );
+	}
+
+	public function test_set_user_notice__sets_new_notice_with_meta() {
+		// Arrange.
+		$user = $this->act_as_admin();
+		$meta = [ 'some-meta-key' => 'some-meta-value' ];
+
+		// Act.
+		User::set_user_notice( 'test_notice', true, $meta );
+
+		// Assert.
+		$notices = get_user_meta( $user->ID, self::ADMIN_NOTICES_KEY, true );
+		$this->assertEquals( [ 'test_notice' => [ 'is_viewed' => true, 'meta' => $meta ] ], $notices );
+	}
+
+	public function test_set_user_notice__keeps_old_notice_meta_when_no_meta_passed() {
+		// Arrange.
+		$user = $this->act_as_admin();
+		$meta = [ 'some-meta-key' => 'some-meta-value' ];
+		$notices = [ 'test_notice' => [ 'is_viewed' => true, 'meta' => $meta ] ];
+
+		update_user_meta( $user->ID, self::ADMIN_NOTICES_KEY, $notices );
+
+		// Act.
+		User::set_user_notice( 'test_notice' );
+
+		// Assert.
+		$notices = get_user_meta( $user->ID, self::ADMIN_NOTICES_KEY, true );
+		$this->assertEquals( [ 'test_notice' => [ 'is_viewed' => true, 'meta' => $meta ] ], $notices );
+	}
+
+	public function test_set_user_notice__updates_notice_meta() {
+		// Arrange.
+		$user = $this->act_as_admin();
+		$meta = [ 'some-meta-key' => 'some-meta-value' ];
+		$notices = [ 'test_notice' => [ 'is_viewed' => true, 'meta' => $meta ] ];
+
+		update_user_meta( $user->ID, self::ADMIN_NOTICES_KEY, $notices );
+
+		// Act.
+		$new_meta = [ 'another-meta-key' => 'another-meta-value' ];
+		User::set_user_notice( 'test_notice', true, $new_meta );
+
+		// Assert.
+		$notices = get_user_meta( $user->ID, self::ADMIN_NOTICES_KEY, true );
+		$this->assertEquals( [ 'test_notice' => [ 'is_viewed' => true, 'meta' => $new_meta ] ], $notices );
+	}
+
+	// BC tests
+	public function test_is_user_notice_viewed__BC__returns_true_when_viewed() {
+		// Arrange.
+		$user = $this->act_as_admin();
+		$notices = [ 'test_notice' => 'true' ];
```

Comment:
'true' as a string on purpose? (same for all below)

Reply:
Yes, this is how it was used to be...